Sofa/Component/SolidMechanics/FEM/Elastic/src/sofa/component/solidmechanics/fem/elastic/TriangularAnisotropicFEMForceField.inl

```diff
@@ -89,24 +89,15 @@ void TriangularAnisotropicFEMForceField<DataTypes>::createTriangleInfo(Index tri
 template< class DataTypes>
 void TriangularAnisotropicFEMForceField<DataTypes>::init()
 {
-    if (l_topology.empty())
-    {
-        msg_info() << "link to Topology container should be set to ensure right behavior. First Topology found in current context will be used.";
-        l_topology.set(this->getContext()->getMeshTopologyLink());
-    }
-
-    m_topology = l_topology.get();
-    msg_info() << "Topology path used: '" << l_topology.getLinkedPath() << "'";
+    Inherited::init();
 
-    if (m_topology == nullptr)
+    if (this->d_componentState.getValue() == sofa::core::objectmodel::ComponentState::Invalid)
     {
-        msg_error() << "No topology component found at path: " << l_topology.getLinkedPath() << ", nor in current context: " << this->getContext()->name;
-        sofa::core::objectmodel::BaseObject::d_componentState.setValue(sofa::core::objectmodel::ComponentState::Invalid);
         return;
     }
 
     // Create specific handler for TriangleData
-    d_localFiberDirection.createTopologyHandler(m_topology);
+    d_localFiberDirection.createTopologyHandler(this->l_topology);
```

Comment:
```suggestion
    d_localFiberDirection.createTopologyHandler(this->l_topology.get());
```
?
not sure about this one...

Reply:
There is an implicit conversion of `Link` to appear as a regular pointer, using the `operator DestType*() const` operator.